src/plots/gl3d/scene.js

```diff
@@ -716,11 +738,121 @@ proto.toImage = function(format) {
 };
 
 proto.setConvert = function() {
-    for(var i = 0; i < 3; ++i) {
+    for(var i = 0; i < 3; i++) {
         var ax = this.fullSceneLayout[axisProperties[i]];
         Axes.setConvert(ax, this.fullLayout);
         ax.setScale = Lib.noop;
     }
 };
 
+proto.updateAnnotations = function() {
+    var anns = this.fullSceneLayout.annotations;
+
+    for(var i = 0; i < anns.length; i++) {
+        mockAnnAxes(anns[i], this);
+    }
+
+    this.fullLayout._infolayer
+        .selectAll('.annotation-' + this.id)
+        .remove();
+};
+
+proto.handleAnnotations = function() {
+    var drawAnnotation = Registry.getComponentMethod('annotations', 'drawRaw');
+    var fullSceneLayout = this.fullSceneLayout;
+    var dataScale = this.dataScale;
+    var anns = fullSceneLayout.annotations;
+    var axLetters = ['x', 'y', 'z'];
+
+    for(var i = 0; i < anns.length; i++) {
+        var ann = anns[i];
+        var annotationIsOffscreen = false;
+
+        for(var j = 0; j < 3; j++) {
+            var axLetter = axLetters[j];
+            var pos = ann[axLetter];
+            var ax = fullSceneLayout[axLetter + 'axis'];
+            var posFraction = ax.r2fraction(pos);
+
+            if(posFraction < 0 || posFraction > 1) {
+                annotationIsOffscreen = true;
+                break;
+            }
+        }
+
+        if(annotationIsOffscreen) {
+            this.fullLayout._infolayer
+                .select('.annotation-' + this.id + '[data-index="' + i + '"]')
+                .remove();
+        } else {
+            ann.pdata = project(this.glplot.cameraParams, [
+                fullSceneLayout.xaxis.r2l(ann.x) * dataScale[0],
+                fullSceneLayout.yaxis.r2l(ann.y) * dataScale[1],
+                fullSceneLayout.zaxis.r2l(ann.z) * dataScale[2]
+            ]);
+
+            drawAnnotation(this.graphDiv, ann, i, ann._xa, ann._ya);
+        }
+    }
+};
+
+function mockAnnAxes(ann, scene) {
+    var fullSceneLayout = scene.fullSceneLayout;
+    var domain = fullSceneLayout.domain;
+    var size = scene.fullLayout._size;
+
+    var base = {
+        // this gets fill in on render
+        pdata: null,
+
+        // to get setConvert to not execute cleanly
+        type: 'linear',
+
+        // don't try to update them on `editable: true`
+        autorange: false,
+
+        // set infinite range so that annotation draw routine
+        // does not try to remove 'outside-range' annotations,
+        // this case is handled in the render loop
+        range: [-Infinity, Infinity]
+    };
+
+    ann._xa = {};
+    Lib.extendFlat(ann._xa, base);
+    Axes.setConvert(ann._xa);
```

Comment:
> what do you need from setConvert, if you're overriding _offset and l2p?

Oh I see - `ax.l2p` is now controlling all the other `2p` conversions (instead of the local var `l2p`) so you get those for free. Seems fine - tiny tiny 🐎  hit from having to look up attributes more often... but should be negligible.